Action Items story. Bug fixes and Enhancement done.

pant-manas opened the pull request:
Worked on bug fixes and enhancements

In bugs there were bugs of onsubmit message and carlisting page coming from middle. Fixed those

In enhancements the error messages were to be changed in form according to the input typed in. 
Like if email input is empty, Email required
If email is incorrect
Invalid Email
Similarly for contact field, name field and address field

https://abhirams.atlassian.net/browse/P3ECWFC-77

https://abhirams.atlassian.net/browse/P3ECWFC-76

Also worked on enhancements
https://abhirams.atlassian.net/browse/P3ECWFC-82
https://abhirams.atlassian.net/browse/P3ECWFC-83

Commit:
spill over work from sprint 1

pashedin commented:
`spill over work from sprint 1`
`Worked on the remaining part of Sprint 1`
these two statements donot give me any idea o what I to expect in the code, mention or list the features as well.

pant-manas commented:
> `spill over work from sprint 1` `Worked on the remaining part of Sprint 1` these two statements donot give me any idea o what I to expect in the code, mention or list the features as well.

@pashedin  there were just enhancements and bugs to fix. So thats why just added the links for the same things.

pashedin commented on src/pages/TestDrive/TestDrive.tsx:
```diff
@@ -1,32 +1,27 @@
+import React from "react";
 import "./TestDrive.scss";
 import { contactpattern, emailpattern } from "../../constants";
+import { useForm } from "react-hook-form";
```
here you are utilizing react-hook form
This PR is removing it Please co-ordiante with Kartik and Pratik
https://github.com/DteTRPod3/TRTraining_FinalAssignment/pull/26/files

pashedin commented on src/pages/TestDrive/TestDrive.tsx:
```diff
@@ -35,85 +30,76 @@ function TestDrive() {
+                      message: "Mobile number cannot be more than 10 digits",
+                    },
+                  })}
                 />
```
why change it back to number, type tel was good, infact much better.
`<input type="tel" placeholder="Mobile number" {...register("Mobile number", {required: true, minLength: 6, maxLength: 12})} />`
https://react-hook-form.com/form-builder/

pashedin commented on src/pages/TestDrive/__test__/TestDrive.test.tsx:
```diff
@@ -52,7 +52,7 @@ describe("Test Drive Form", () => {
 
     expect(screen.getByPlaceholderText(/enter contact\.\.\./i)).toHaveValue(
-      "1111111111"
+      1111111111
```
For test cases add believable input 😄 
1234567890 is still fine, or I would say add a random but believable input 9923876104.
also add the test cases where we test that on submitting wrong value an error is thrown.

Commit:
Pr reviews and comments worked on

pant-manas commented:
> Referen

@pashedin  Commited the changes you suggested.
Added test case for error messages
Added tel for attribute for contact
Completed the bugs and enhancements

pashedin reviewed (approved):
Nice quick work man 👍
